pydantic/generics.py

```diff
@@ -61,12 +75,10 @@ def __class_getitem__(cls: Type[GenericModelT], params: Union[Type[Any], Tuple[T
             ),
         )
         created_model.Config = cls.Config
-        concrete = all(not _has_typevar(v) for v in concrete_type_hints.values())
+        concrete = not any(_generate_parameters(concrete_type_hints.values()))
         created_model.__concrete__ = concrete
         if not concrete:
-            parameters = tuple(v for v in concrete_type_hints.values() if _has_typevar(v))
-            parameters = tuple({k: None for k in parameters}.keys())  # get unique params while maintaining order
-            created_model.__parameters__ = parameters
+            created_model.__parameters__ = _extract_parameters(typevars_map)
```

Comment:
Minor optimization:

```
new_params = _extract_parameters(typevars_map)
created_model.__concrete__ = bool(new_parameters)
created_model.__parameters__ = new_params
```

Reply:
correct, also removed extract parameters as it is now only used once and moved a test to it's counterpart that already existed